Doc/library/optparse.rst

```diff
@@ -10,18 +10,47 @@
 
 **Source code:** :source:`Lib/optparse.py`
 
-.. deprecated:: 3.2
-   The :mod:`optparse` module is :term:`soft deprecated` and will not be
-   developed further; development will continue with the :mod:`argparse`
-   module.
+.. note::
+
+   :mod:`argparse` (rather than this module) is the recommended standard
+   library module for implementing command line applications unless one
+   of the following caveats applies:
+
+   * the application requires additional control over the way options and
+     positional parameters are interleaved on the command line (including
+     the ability to disable the interleaving feature completely)
+   * the application requires additional control over the incremental parsing
+     of command line elements (while ``argparse`` does support this, the
+     exact way it works in practice is undesirable for some use cases)
+   * the application requires additional control over the handling of options
+     which accept parameter values that may start with ``-`` (such as delegated
+     options to be passed to invoked subprocesses).
+
+   These ``argparse`` caveats also mean that :mod:`optparse` is likely to
+   provide a better foundation for library authors *writing* third party
+   command line argument processing libraries.
+
+.. seealso::
+
+    The :pypi:`"click" package <click>` is an ``optparse`` based third party
+    argument processing library which allows command line applications to be
+    developed as a set of appropriately decorated command implementation
+    functions.
+
+.. seealso::
+
+    The :pypi:`"Typer" package <click>` is a ``click`` based third party
+    argument processing library which allows the use of annotated Python
+    type hints to define an application's command line interface.
 
 --------------
 
 :mod:`optparse` is a more convenient, flexible, and powerful library for parsing
-command-line options than the old :mod:`getopt` module.  :mod:`optparse` uses a
-more declarative style of command-line parsing: you create an instance of
-:class:`OptionParser`, populate it with options, and parse the command
-line. :mod:`optparse` allows users to specify options in the conventional
+command-line options than the minimalist :mod:`getopt` module.
```

Comment:
Replacing `old` with `minimalist` is the only actual wording change here (the rest is just reflowing text)